src/Standards/Generic/Sniffs/Files/FilePermissionsSniff.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Tests that files are not executable.
+ *
+ * @author    Matthew Peveler <matt.peveler@gmail.com>
+ * @copyright 2019 Matthew Peveler
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\Files;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+
+class FilePermissionsSniff implements Sniff
+{
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [T_OPEN_TAG];
```

Comment:
You may want to add `T_OPEN_TAG_WITH_ECHO` as well to also examine file which only use the short echo PHP tag.

Reply:
Most of the other File standards seem to use a similar register. Would it make sense to also update those to also have that tag? I'm fine with just adding it here though.

Reply:
I think that depends on the individual sniff and should probably be done in a separate PR in that case.

Reply:
Within just Generic, it includes all the other sniffs that test for the beginning of a file:
* EndFileNewlineSniff
* EndFileNoNewlineSniff
* LineEndingsSniff
* LineLengthSniff
* LowercasedFilenameSniff

Agreed that fixing them would be a separate PR, just trying to clarify what the expected style though as I'm not well-versed enough in this project to know when to use one or both.

Reply:
Without looking at the sniffs you've listed - IMO the consideration here would be "_does this rule apply to all PHP files or just to PHP-only files ?_"

If the rule applies only to PHP-only files, then don't add it.
If the rule applies to PHP-only files as well as mixed HTML/PHP files, then yes, it should be added.